backend/capellacollab/projects/toolmodels/backups/routes.py

```diff
@@ -199,8 +200,14 @@ def get_logs(
     pipeline: DatabaseBackup = Depends(injectables.get_existing_pipeline),
 ):
     backup: Backup = Backup.from_orm(pipeline)
+    if not backup.lastrun:
+        raise fastapi.HTTPException(
+            status_code=404,
+            detail={
+                "err_code": "PIPELINES_NO_LAST_RUN",
+                "reason": "There is no last run available for the pipelines. Please trigger the pipeline first.",
+            },
+        )
 
-    logs = ""
-    if backup.lastrun and backup.lastrun.id:
-        logs = OPERATOR.get_job_logs(id=backup.lastrun.id)
+    logs = operators.get_operator().get_job_logs(id=backup.lastrun.id)
     return helper.filter_logs(logs, [pipeline.t4c_password])
```

Comment:
> I am a bit confused about the `get_job_logs` because it either returns the logs or when an exception occurs it returns the events retrieved by some `field_selector`. So the name `get_job_logs` is somehow miss leading and I would definitely adjust it.
> Furthermore, I am confused that this here actually worked at some point because `get_job_logs` expects an argument `_id` and no longer `id`. So I think I might have forgotten to changes this in a previous PR.

Fixed

> Lastly, the types are wrong here since `backup.lastrun.id` is potentially `None` but `id`/`_id` expects a string.

Interesting. I've changed the type to `str`. The only initialisation of `BackupJob` has an id that is never None.